File: 20-graphs2/1-intro/main.cpp
// Advanced Graph Algorithms Introduction - C++ Example

#include <iostream>
#include <unordered_map>
#include <vector>

int main() {
    std::unordered_map<int, std::vector<int>> graph;
    graph[0] = {1, 2};
    graph[11] = {0};
    graph[10] = {0, 3};
    graph[12] = {2};

    std::cout << "Graph vertices: ";
    for (auto& pair : graph) std::cout << pair.first << " ";
    std::cout << "\nGraph edges from 0: ";
    for (int v : graph) std::cout << v << " ";
    std::cout << std::endl;

    return 0;
}

File: 20-graphs2/1-intro/main.java
// Advanced Graph Algorithms Introduction - Java Example

import java.util.*;

class GraphIntroAdvanced {
    public static void main(String[] args) {
        Map<Integer, List<Integer>> graph = new HashMap<>();
        graph.put(0, Arrays.asList(1, 2));
        graph.put(1, Arrays.asList(0));
        graph.put(2, Arrays.asList(0, 3));
        graph.put(3, Arrays.asList(2));

        System.out.println("Graph vertices: " + graph.keySet());
        System.out.println("Graph edges from 0: " + graph.get(0));
    }
}
